gmime/envelope.go

```diff
@@ -69,6 +69,32 @@ func (m *Envelope) Headers() textproto.MIMEHeader {
 	return goHeaders
 }
 
+// ReplaceHeader replaces the header with matching key & originalValue with replaceValue
+func (m *Envelope) ReplaceHeader(key, originalValue, replaceValue string) error {
+	headers := C.g_mime_object_get_header_list(m.asGMimeObject())
+	count := C.g_mime_header_list_get_count(headers)
+	var i C.int
+	for i = 0; i < count; i++ {
+		header := C.g_mime_header_list_get_header_at(headers, i)
+		name := C.GoString(C.g_mime_header_get_name(header))
```

Comment:
Do we need to free any of these strings?

Reply:
i think only CString needs to be freed

Reply:
Doesn't this line create a C string then convert it to a Go string?

Reply:
ahh. yeah, these are part of headerList so gmime takes care of it.
https://developer.gnome.org/gmime/stable/GMimeObject.html#g-mime-object-get-header-list
> the GMimeHeaderList for object . Do not free this pointer when you are done with it.

Reply:
What about: g_mime_header_get_name?

Reply:
it returns `header->name`, so I assume as gmime frees headerList, name is also freed?

Reply:
in that case probably.